Restrained-Electrostatic-Potential/espfit.py

```diff
@@ -0,0 +1,322 @@
+            resp_b: float; restraint parabola tightness b
+            ihfree: bool: hydrogen excluded or included in restraint
+            symbols: string array of element symbols
+            n_atoms: list of the number of atoms in each molecule
+            n_constraints: list of the number of constraints for each molecule
+       output:
+            a: restrained A array
+    """
+
+    # hyperbolic Restraint
+    a = copy.deepcopy(akeep)
+    n_mol = len(n_atoms)
+    index = 0
+    index_q = 0
+    for mol in range(n_mol):
+        for i in range(n_atoms[mol]):
+            if not ihfree[mol] or symbols[mol][i] != 'H':
+                a[index+i, index+i] = akeep[index+i, index+i]\
+                + resp_a[mol]/np.sqrt(q[index_q]**2 + resp_b[mol]**2)
+            index_q += 1
+        index += n_atoms[mol] + n_constraints[mol]
+    return a
+
+def iterate(q, akeep, b, resp_a, resp_b, ihfree, symbols, toler,\
+            maxit, n_atoms, n_constraints, indecies):
+    """Function to iterate the RESP fitting procedure
+       input:
+                q: array of initial charges
+                akeep: array of unrestrained A matrix
+                b: array of matrix B
+                resp_a: float; restraint scale a
+                resp_b: float; restraint parabola tightness b
+                ihfree: bool: hydrogen excluded or included in restraint
+                symbols: array of element symbols
+                toler: float; tolerance for charges in the fitting
+                maxit: int; maximum number of iterations
+                n_atoms: list of the number of atoms in each molecule
+                n_constraints: list of the number of constraints for
+                                each molecule
+                indecies: an array of the indecies for the atoms in the
+                          A and B matrices
+        output:
+                q: array of fitted charges
+    """
+    n_mols = len(n_atoms)
+    qkeep = q[indecies]
+    niter = 0
+    difm = 1
+    while difm > toler and niter < maxit:
+        index = 0
+        niter += 1
+        a = restraint(q[indecies], akeep, resp_a, resp_b, ihfree,\
+                      symbols, n_atoms, n_constraints)
+        q, note = esp_solve(a, b)
+        q_q = q[indecies]
+        difm = 0
+            
+        for i in range(len(q_q)):
+            dif = (q_q[i]-qkeep[i])**2
+            if difm < dif:
+                difm = dif
+        qkeep = copy.deepcopy(q_q)
+        difm = np.sqrt(difm)
+    if difm > toler:
+        note += '\nCharge fitting did not converge; ' +\
+               'try increasing the maximum number of iterations to ' +\
+               '> %i.' %maxit
+    return q_q, note
+
+def intramolecular_constraints(constraint_charge, constraint_equal, constraint_groups):
+    """Function to extract intramolecular constraints from user constraint input
+       input:
+             constraint_charge: list of lists of charges and atom indecies list
```

Comment:
Can you use [numpy style docs](http://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_numpy.html) for the whole repository?